src/main/java/uk/gov/hmcts/reform/draftstore/service/crypto/CryptoService.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.draftstore.service.crypto;
+
+import com.google.common.base.Strings;
+import com.google.common.primitives.Bytes;
+import uk.gov.hmcts.reform.draftstore.utils.IllegalArgument;
+
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import static com.google.common.base.Charsets.UTF_8;
+
+public class CryptoService {
+
+    private static final String CIPHER = "AES/GCM/NoPadding";
+    private static final Length KEY_LENGTH = Length.BITS_128;
+    private static final Length IV_LENGTH = Length.BITS_96;
+    private static final Length GMC_TAG_LENGTH = Length.BITS_128;
+
+    public static byte[] encrypt(String input, String password) {
+
+        IllegalArgument.throwIf(() -> input == null, "Input can't be null");
+        IllegalArgument.throwIf(() -> Strings.isNullOrEmpty(password), "Password can't be empty");
```

Comment:
Any requirements on the password?

Reply:
It would be a good idea add some validation of the password (length) to request validation. Not critical, though, as our clients are other HMCTS services (responsible for using strong passwords), not individuals.

Reply:
probably a min length would be enough...

Reply:
I'd rather validate that on API level

Reply:
Alternatively we can require a 128bit key to be sent from the client. Instead of deriving it from the password

Reply:
Those passwords could be named `secrets`. To us they're secret pieces of information, based on which we will encrypt the data.